protocols/v2/binary-sv2/no-serde-sv2/derive_codec/src/lib.rs

```diff
@@ -29,6 +96,43 @@ fn is_already_sized(item: TokenStream) -> bool {
     }
     false
 }
+
+// Filters out attributes from a `TokenStream` that are prefixed with `#`.
+//
+// Removes all Rust attributes (e.g., `#[derive(...)]`, `#[cfg(...)]`)
+// from the provided `TokenStream`, leaving behind only the core structure and
+// its fields. This is useful in procedural macros when you want to manipulate
+// the underlying data without the attributes.
+//
+// # Example
```

Comment:
I added explicit documentation for these methods because unit tests cannot be added for them in the procedural module, and simply reading the methods isn’t very intuitive. Providing a concrete example makes them easier to understand. Currently, we’re using the vanilla token stream to parse the structs, which might be challenging to interpret or modify for someone with limited knowledge of ASTs

Reply:
I added explicit documentation for these methods because unit tests cannot be added for them in the procedural module, and simply reading the methods isn’t very intuitive. Providing a concrete example makes them easier to understand. Currently, we’re using the vanilla token stream to parse the structs, which might be challenging to interpret or modify for someone with limited knowledge of ASTs